src/charm.py

```diff
@@ -566,13 +566,8 @@ def _generate_per_app_config(self, data: "RequirerData_IPA") -> Tuple[dict, str]
             }
         }
 
-        middlewares = self._generate_middleware_config(data, prefix)
-        if middlewares:
-            router_cfg["middlewares"] = list(middlewares.keys())
```

Comment:
The difference here is pretty marginal, but if you look below, the additional selector of `traefik_router_name` is necessary for this to actually work.